test/perf/dotnet-tools/BenchmarkDotNet.Extensions/TooManyTestCasesValidator.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Validators;
+
+namespace BenchmarkDotNet.Extensions
+{
+    /// <summary>
+    /// we need to tell our users that having more than 16 test cases per benchmark is a VERY BAD idea
+    /// </summary>
+    public class TooManyTestCasesValidator : IValidator
```

Comment:
I'm not very clear about the purpose of this validator.
My understanding of a benchmark is something like "Engine.Scripting.InvokeMethod" as listed by `--list flat`. For this specific benchmark, it uses the `ParamsSource` attribute for the `Script` property with 4 values, so this benchmark actually maps to 4 tests. So, this validator means, for this specific benchmark, I should not have more than 16 values defined for the `ParamsSource`, is this understanding correct?

Reply:
This validator ensures that every benchmark has no more than 16 unique sets of parameters. In the past we had issues with `[ArgumentsSource]` and `[ParamsSource]` which were very often copy-pasted from some unit-test projects that were testing hundreds of test cases. Example:

```cs
[Benchmark]
[ArgumentsSource(nameof(Numbers))]
public double ManyArguments(double x, double y) => Math.Pow(x, y);

public IEnumerable<object[]> Numbers() // for multiple arguments it's an IEnumerable of array of objects (object[])
{
    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < 10; j++)
        {
            yield return new object[] { (double)i, (double)j };
        }
    }
}
```

Here `ManyArguments` has 10x10 unique sets of arguments, which gives us 100 benchmarks runs. Every run is 3-5s on average, so we end up needing 5-8.33 minutes just to run a single benchmark.

Reply:
Clear, thanks! I think this validator would be useful in general, especially as you said, it may be common to copy unit tests and wrap them to benchmarks.